File: Java/src/Find Minimum in Rotated Sorted Array II.java
public class Solution {
    public int findMin(int[] nums) {
        int start = 0;
        int end = nums.length-1;
        int mid;
        
        while (start < end){
            mid = (start+end) / 2;
            if (nums[mid] > nums[end]){
                start = mid + 1;
            }
            else if (nums[mid] < nums[end]){
                end = mid;
            }
            else{
                end--;
            }
        }
        return nums[start];
    }
}

File: Python/Find Minimum in Rotated Sorted Array II.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def findMin(self, nums):
        start = 0
        end = len(nums) - 1
        while start < end:
            mid = (start+ end) / 2
            if nums[mid] > nums[end]:
                start = mid + 1
            elif nums[mid] < nums[end]:
                end = mid
            else:
                end -= 1
        return nums[start]
                